src/main/java/org/testng/internal/ExitCodeListener.java

```diff
@@ -63,4 +70,11 @@ public void onStart(ITestContext context) {}
 
   @Override
   public void onFinish(ITestContext context) {}
+
+  @Override
+  public void onExecutionFinish() {
+    if (failIfAllTestsSkipped && (getStatus().getExitCode() == ExitCode.SKIPPED)) {
```

Comment:
Why not failing without success?

Reply:
I didn't quite get you @juherr . Can you please tell me what you are hinting at ?

Reply:
I mean for all exit code != 0

Reply:
@juherr - I think only in the case of all skipped tests (due to a failure in the data provider) we don't bubble out any exceptions. In all other cases (failed tests, skipped tests due to configuration failures) we are good and the build is already getting failed (else we would have had people complain about this already)

All tests getting skipped due to a failure in the data provider is the exception flow that I am trying to address via this PR

Reply:
ok